lang_qc/models/pacbio/experiment.py

```diff
@@ -0,0 +1,143 @@
+        """,
+    )
+    study_name: str = Field(
+        default=None,
+        title="Study name",
+        description="""
+        Study name, is not set in case of multiple studies.
+        """,
+    )
+    sample_id: str = Field(
+        default=None,
+        title="Sample identifier",
+        description="""
+        Sample identifier, is not set in case of multiple samples.
+        """,
+    )
+    sample_name: str = Field(
+        default=None,
+        title="Sample name",
+        description="""
+        Sample name, is not set in case of multiple samples.
+        """,
+    )
+    num_samples: int = Field(
+        default=1,
+        title="Number of samples",
+        description="""
+        Number of samples. If the data is for a well, multiple
+        samples might be present.
+        """,
+    )
+    tag_sequence: list = Field(
+        title="Tag sequence",
+        description="""
+        Tag sequences as a list. The list is empty in case of multiple
+        samples or if the library for the only present sample does not
+        have the index sequence.
+        """,
+    )
+    library_type: list = Field(
+        title="Library type",
+        description="""
+        Library types as a sorted list of unique strings to cover an
+        unlikely case of multiple library types.
+        """,
+    )
+
+    class Config:
+        orm_mode = True
+        extra = Extra.forbid
+
+    @classmethod
+    def from_orm(cls, lims_db_rows: List[PacBioRun]):
+        """
+        A factory method, creates an instance of the PacBioLimsData class.
+        Should be given a non-empty list of PacBioRun table row objects as
+        an argument.
+        """
+
+        num_samples = len(lims_db_rows)
+        if num_samples == 0:
+            raise Exception("Cannot create PacBioLimsData object, no data.")
+        if any(row is None for row in lims_db_rows):
+            raise Exception("Cannot create PacBioLimsData object, None row.")
+
+        # Using sets for some data instead of lists because we do not
+        # want repetitions.
+        lims_data = {
+            "num_samples": num_samples,
+            "study_id": set(),
+            "library_type": set(),
+            "tag_sequence": [],
+        }
+        for row in lims_db_rows:
+            lims_data["study_id"].add(row.study.id_study_lims)
+            lims_data["library_type"].add(row.pipeline_id_lims)
+            if num_samples == 1:
+                tag = row.tag_sequence
+                if tag:
+                    lims_data["tag_sequence"].append(tag)
+                    tag = row.tag2_sequence
+                    if tag:
+                        lims_data["tag_sequence"].append(tag)
+                lims_data["sample_id"] = row.sample.id_sample_lims
+                lims_data["sample_name"] = row.sample.name
+                lims_data["study_name"] = row.study.name
+
+        # Convert sets back to lists and sort so that the list items are
```

Comment:
Maybe just use `sorted()` function on your sets? Then you get the list you want without casting yourself

Reply:
Good tip, thanks, done.